### PR TITLE
Add global state to wallet (decouple web3-react)

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

This PR adds the state of the wallet as a joitai state. 

This way it the app don't need to depend from any wallet library anymore. It will use `useWalletInfo` and `useWalletDetails` hooks, which just read from the global state.

There's a new updater, which deals with the wallet library itself. 

Here is an example of autoconnecting to Metamask and then connecting to Impersonator using Wallet Connect. Because the updater memorize the values, it should not update the joitai state too often.

image

# To Test

Try to connect using some representing wallets to make sure I didn't break any important use case.

### CODE REVIEW

File: src/cow-react/modules/wallet/index.ts [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
 export * from './web3-react/hooks/useDisconnectWallet'
+export * from './web3-react/hooks/useWalletMetadata'
+export * from './web3-react/hooks/useEagerlyConnect'
+export * from './web3-react/hooks/useOrderedConnections'
 
+// Updater
+export * from './web3-react/updater'
+
+// Components
 export * from './web3-react/containers/Web3Status'
```
Comment (anxolin) [inline src/cow-react/modules/wallet/index.ts:14]: This `index` is serving the porpouse of a facade of the module.

Everythin that is exposed here has a simple interface that don't depend in web3-react. 

This way, we can import from `@cow/modules/wallet` all relevant things, like the updater, hooks, or types

File: src/cow-react/modules/wallet/web3-react/hooks/useWalletMetadata.ts [hunk 2]
```diff
@@ -59,36 +60,11 @@ export function useWalletMetaData(): WalletMetaData {
...
 
     if (connectionType === ConnectionType.GNOSIS_SAFE) {
-      return {
-        walletName: GNOSIS_SAFE_APP_NAME,
-        icon: SAFE_ICON_URL,
-      }
+      return METADATA_SAFE
     }
 
-    return { walletName: '', icon: '' }
```
Comment (anxolin) [inline src/cow-react/modules/wallet/web3-react/hooks/useWalletMetadata.ts:68]: ATENTION: I'm a bit unsure why before this was returning empty strings. 

The type allows undefined. So I guess it should return that when is unknown.

Comment (shoom3301) [reply src/cow-react/modules/wallet/web3-react/hooks/useWalletMetadata.ts:68]: I think there are no special reasons, it should be undefined

**Review by shoom3301 — APPROVED** [review]

**Review by elena-zh — APPROVED** [review]
Hey @anxolin , great job!
No new issues with connect wallet functionality besides known ones as #2070 and issues reported in https://github.com/cowprotocol/cowswap/pull/2044#issuecomment-1441896931 (but they were fixed later), and #2022

**Pull request merged**